### PR TITLE
Generate spark sql based on the jdbc datasource and optimized code

### PR DESCRIPTION
**Pull request by ChengJie1053**
Generate spark sql based on the jdbc datasource，the response is shown below

image

### CODE REVIEW

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/common/src/main/java/org/apache/linkis/metadata/query/common/service/AbstractDbMetaService.java [hunk 1]
```diff
@@ -67,6 +68,12 @@ public List<MetaColumnInfo> getColumns(
     return this.getConnAndRun(operator, params, conn -> this.queryColumns(conn, database, table));
   }
 
+  @Override
+  public GenerateSqlInfo getSparkSql(
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/common/src/main/java/org/apache/linkis/metadata/query/common/service/AbstractDbMetaService.java:72]: It is not recommended to add Spark-related methods to AbstractDbMetaService, and a new interface should be added

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/AbstractSqlConnection.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      }
+      return primaryKeys;
+    } finally {
+      if (null != rs) {
+        rs.close();
+      }
+    }
+  }
+
+  public GenerateSqlInfo getSparkSql(String database, String table) {
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/AbstractSqlConnection.java:116]: It is not recommended to add Spark-related methods here, and a new interface should be added

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/postgres/SqlConnection.java [hunk 3]
```diff
@@ -99,126 +85,30 @@ public List<String> getAllTables(String schemaname) throws SQLException {
...
    * @param connectMessage
    * @param database
    * @return
    * @throws ClassNotFoundException
    */
-  private Connection getDBConnection(ConnectMessage connectMessage, String database)
+  public Connection getDBConnection(ConnectMessage connectMessage, String database)
       throws ClassNotFoundException, SQLException {
+    String extraParamString =
+        connectMessage.extraParams.entrySet().stream()
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/postgres/SqlConnection.java:97]: Need to judge non-empty execution?

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/common/src/main/java/org/apache/linkis/metadata/query/common/service/AbstractDbMetaService.java (hunk 1 above)
Comment (ChengJie1053) [reply linkis-public-enhancements/linkis-datasource/linkis-metadata-query/common/src/main/java/org/apache/linkis/metadata/query/common/service/AbstractDbMetaService.java:72]: Ok, thank you for reviewing the code

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/postgres/SqlConnection.java (hunk 3 above)
Comment (ChengJie1053) [reply linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/postgres/SqlConnection.java:97]: Ok, thank you for reviewing the code

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**